include/aws/s3/s3_client.h

```diff
@@ -632,6 +632,14 @@ struct aws_s3_meta_request_options {
      * Do not set the message's body-stream if the body is being passed by other means (see note above) */
     struct aws_http_message *message;
 
+    /**
+     * Optional.
+     * If set, the received data will be written into this file.
+     * the `body_callback` will still be invoked if set.
```

Comment:
Do we have a use case where we might need both? I think it would be simpler to return an error if both are set until we have a use case for this. You can upload an object and then call the getObject in the tests instead of relying on the body callback.

Reply:
there is no specific reason to not invoke the body callback. It's up to the user to decide what to do with the response body. 
I think it's better to not assume what will be the use case for user.

Reply:
We can discuss it, I think it's better to keep it restricted if we want to optimize it in the future, such as out-of-order writes or something. This will restrict almost all future optimizations because we will still need to trigger the body callback.

Reply:
I think we can still do future opt in the future by add new options. But, since we always default to some body callback in the binding, it maybe a performance concern.